core/eolearn/core/eodata_io.py

```diff
@@ -156,41 +170,35 @@ def _save_single_feature(save_spec: tuple[type[FeatureIO[T]], T, str], *, filesy
 
 def remove_redundant_files(
     filesystem: FS,
-    eopatch_features: list[FeatureSpec],
+    data_for_saving: list[tuple[type[FeatureIO], Any, str]],
     preexisting_files: FilesystemDataInfo,
     current_compress_level: int,
 ) -> None:
-    """Removes files that should have been overwritten but were not due to different compression levels."""
-
-    def has_different_compression(path: str | None) -> bool:
-        return path is not None and MimeType.GZIP.matches_extension(path) != (current_compress_level > 0)
+    """Removes files that should have been overwritten but were not due to different file extensions."""
+    feature_paths = (path for _, ftype_dict in preexisting_files.features.items() for _, path in ftype_dict.items())
+    meta_paths = (preexisting_files.bbox, preexisting_files.meta_info, preexisting_files.timestamps)
+    split_paths = (path.split(".", maxsplit=1) for path in filter(None, itertools.chain(meta_paths, feature_paths)))
```

Comment:
What if a `.` occurs in the path before the extension? sounds like it could be problematic, but not sure if anyone would go and persist with such a choice.

Reply:
i also had such concerns
I feel like I'll have to write a custom `split_ext` function that takes into account multi-extensions.

Reply:
It could be done with regex and [lookahead](https://www.geeksforgeeks.org/python-regex-lookahead/) `(?==)`

```python
import re
complex_path = r'a\b/c.d\f.g/e.np.gzip'
re_ext = re.compile(r'\.(?=[^/\\]*$)')
```

The above searches for a match where there is a literal dot, and no path characters (`/` or the escaped `\` for windows) after the dot, then performs a single split.

```
In [49]:  re_ext.split(s, maxsplit=1)
Out[49]: ['a\\b/c.d\\f.g/e', 'np.gzip']
```

Reply:
does it handle dot-hidden files? :)
i prefer to avoid regex when possible since its hard to debug.
I'm cooking up a solution with pathlib, which is stronger than os.path or fs.path

Reply:
done